DEVELOPERS.md

```diff
@@ -269,74 +288,82 @@ pytest -m "workflow_test and not external_workflow_test"
 pytest -m "not external_workflow_test"
 ```
 
+## Understanding the pixelator plugin system
+Pixelator has a plugin system that allows you to extend the functionality of pixelator.
+This section outlines how to create new plugins and make sure these are picked-up
+by pixelator at runtime.
 
+### CLI plugins
 
-## Utility scripts
+Pixelator has a simple plugin system that allows you as a developer to add groups
+of commands to the pixelator command-line interface.
 
-Pixelator provides a number of utility scripts to help with development and testing using [Task](https://taskfile.dev/).
+This uses the ability of the python package `Click https://click.palletsprojects.com/en/8.1.x/`_
+to add groups of commands to the command line interface. To write a plugin you need to define a
+group, below is an example of a new group `my_plugin` with a single command `my_command`:
+
+```python
+import sys
 
-Installation instruction for can be found [here](https://taskfile.dev/#/installation).
+import click
 
-> [!TIP]
-> Depending on the installation method the task executable might be named `task` or `go-task`.
 
-### Viewing the taskfile
+@click.group()
+def my_cli_plugin():
+    """My custom plugin"""
+    pass
 
-Run the following command to view the available tasks:
 
-```shell
-task --list
-```
-```console
-* coverage:                            Run tests using pytest and generate a coverage report.
-* format:                              Format code using black.
-* format-check:                        Check code formatting using ruff.
-* lint:                                Run linting using ruff.
-* test:                                Run tests using pytest with the default flags defined in pyproject.toml.
-* test-all:                            Run all tests using pytest.
-* test-nf-core-pixelator:              Run the default nf-core/pixelator test profile with this version of pixelator.
-* test-web:                            Run web tests using pytest.
-* test-workflow:                       Run workflow tests using pytest.
-* test-workflow-external:              Run external workflow tests using pytest.
-* typecheck:                           Run type checking using mypy.
-* tests:update-report-test-data:       Update the report test data using the nf-core/pixelator test profile.
-* tests:update-web-test-data:          Create web test data.
-```
+@click.command()
+def my_command():
+    """My custom command"""
+    click.echo("Hello!")
 
-View more detailed documentation for a specific task:
 
-```shell
-task test-nf-core-pixelator --summary
+my_cli_plugin.add_command(my_command)
+
+
+if __name__ == "__main__":
+    sys.exit(my_cli_plugin())
 ```
-```console
-task: test-nf-core-pixelator
 
-Run the default nf-core/pixelator test profile with this version of pixelator.
+For pixelator to find this you need to define an `entrypoint https://setuptools.pypa.io/en/latest/userguide/entry_point.html#entry-points-for-plugins`_.
```

Comment:
Same here as before - this might be also ancilliary text?

```suggestion
For pixelator to find this you need to define an [entrypoint](https://setuptools.pypa.io/en/latest/userguide/entry_point.html#entry-points-for-plugins).
```